client/consul/consul.go

```diff
@@ -2,18 +2,14 @@ package consul
 
 import (
 	"github.com/hashicorp/nomad/command/agent/consul"
-	"github.com/hashicorp/nomad/nomad/structs"
 )
 
 // ConsulServiceAPI is the interface the Nomad Client uses to register and
 // remove services and checks from Consul.
 type ConsulServiceAPI interface {
-	RegisterGroup(*structs.Allocation) error
-	RemoveGroup(*structs.Allocation) error
-	UpdateGroup(oldAlloc, newAlloc *structs.Allocation) error
-	RegisterTask(*consul.TaskServices) error
-	RemoveTask(*consul.TaskServices)
-	UpdateTask(old, newTask *consul.TaskServices) error
+	RegisterWorkload(*consul.WorkloadServices) error
+	RemoveWorkload(*consul.WorkloadServices)
+	UpdateWorkload(old, newTask *consul.WorkloadServices) error
```

Comment:
I really like the reduction in information sharing we're doing here by passing in this `consul.WorkloadServices` rather than the whole `structs.Allocation`. 👍